File: javapatterns/src/main/java/org/iut/nantes/state/car/Arret.java
package org.iut.nantes.state.car;

class Arret extends Etat
{
  public Arret( GoogleCar car )
  {
    super( car );
  }

  public void bouton()
  {
    System.out.println("Bouton appuyé lors de l'état d'arrêt : on démarre le moteur.");
    car.setEtat( car.getAttente() );
  }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/state/car/etats/Arret.kt
package org.iut.nantes.state.car.etats

import org.iut.nantes.state.car.Car

class Arret(car: Car) : Etat(car) {

    override fun action() {
        println("Voiture à l'arrêt.")
    }

    override fun bouton() {
        println("Démarrage de l'automobile")
        car.changeEtatAttente()
    }


}